tanjun/annotations.py

```diff
@@ -90,8 +90,21 @@
 from .commands import slash
 
 if typing.TYPE_CHECKING:
+    import typing_extensions
     from typing_extensions import Self
 
+    _P = typing_extensions.ParamSpec("_P")
+    # TODO: test this
+    _ConverterValueT = typing.TypeVar("_ConverterValueT", bound=int | str | float)
+    __ConverterSig = (
+        collections.Callable[
+            typing_extensions.Concatenate[_ConverterValueT, _P],
+            collections.Coroutine[typing.Any, typing.Any, typing.Any],
+        ]
+        | collections.Callable[typing_extensions.Concatenate[_ConverterValueT, _P], typing.Any]
```

Comment:
Consider switching to a union in the return statement rather than a union of callables since the distinction doesn't matter anymore